docs/require-by-string-relative-paths.md

```diff
@@ -0,0 +1,169 @@
+
+We need to add relative paths to `require` statements to facilitate the grouping together of related Luau files into libraries and allow for future package managers to be developed and integrated easily.
+
+## Motivation
+
+The Roblox engine does not currently support require-by-string. One motivation for this RFC is to consolidate require syntax and functionality between the Roblox engine and the broader Luau language itself.
+
+Luau itself currently supports a basic require-by-string syntax that allows for requiring Luau modules by relative or absolute path. Unfortunately, the current implementation has a few issues.
+
+### Relative paths
+
+Currently, relative paths are always evaluated relative to the current working directory that the Luau CLI is running from. This leads to unexpected behavior when requiring modules from "incorrect" working directories.
+
+Suppose the module `math.luau` is located in `/Users/JohnDoe/LuauModules/Math` and contains the following:
+
+```lua
+-- Beginning of /Users/JohnDoe/LuauModules/Math/math.luau
+local sqrt = require("../MathHelperFunctions/sqrt")
+```
+
+If we then launched the Luau CLI from the directory `/Users/JohnDoe/Projects/MyCalculator` and required `math.luau` as follows:
+
+```lua
+> local math = require("/Users/JohnDoe/LuauModules/Math/math")
+```
+
+This would cause the following:
+- The current implementation of require would successfully find `math.luau`, as its absolute path was given.
+- Then, it would execute the contents of `math.luau` from the context of the current working directory `/Users/JohnDoe/Projects/MyCalculator`.
+- When attempting to require `sqrt.luau` from `math.luau`, instead of looking in the directory `/Users/JohnDoe/LuauModules/MathHelperFunctions`, the relative path will be evaluated in relation to the current working directory.
+- This will look for `sqrt.luau` in `/Users/JohnDoe/Projects/MathHelperFunctions`, which is a directory that may not even exist.
+
+This behavior is [problematic](https://github.com/Roblox/luau/issues/959), and puts an unnecessary emphasis on the directory from which the Luau CLI is running. A better solution is to evaluate relative paths in relation to the file that is requiring them.
+
+
+### Package management
+
+While package management itself is outside of the scope of this RFC, we want to make it possible for a package management solution to be developed in a way that integrates well with our require syntax.
+
+To require a Luau module under the current implementation, we must require it either by relative or absolute path:
+
+#### Relative paths
+
+Modules can be required relative to the requiring file's location in the filesystem (note, this is different from the current implementation, which evaluates all relative paths in relation to the current working directory).
+
+If we are trying to require a module called `MyModule.luau` in `C:/MyLibrary`:
+```lua
+-- From C:/MyLibrary/SomeModule.luau
+local MyModule = require("MyModule")
```

Comment:
In the other RFC for require by alias, there are bits of examples that says `local Roact = require("Roact")` which resolves to the package `Roact`, but then you say this is a relative require.

I propose that we don't support this either. Require an explicit `./` or `../`.

If you have some file named `Foo` for which you say `require("Foo")`, and you happen to install some library named `Foo`, you now have to define the _order_ of which one should be required. Is it the file, or the library?

Similarly, if you already have the package named `Bar`, and you added a file named `Bar`, you might end up changing the other `require("Bar")` in the sibling files to resolve to the wrong module.

By demanding an explicit `./`/`../`, you avoid the above problem.

Reply:
Yep, that makes sense to me. It's also just more readable even if it is unambiguous.

Reply:
I understand the concern with the ambiguity but I think it is at the cost of usability. My preference would be to use prefixes to disambiguate requires but have no-prefix available as a short-hand with a defined order that we perform the lookup in.

Regardless, I don't think this matters for this RFC. We can always start more restrictive and relax things later. The alias RFC can cover this.